tests/functests/instancetype_test.go

```diff
@@ -0,0 +1,135 @@
+package tests
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "kubevirt.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/utils/pointer"
+	instancetypev1beta1 "kubevirt.io/api/instancetype/v1beta1"
+)
+
+var _ = Describe("Common instance types func tests", func() {
+
+	Context("VirtualMachine using a preference with resource requirements", func() {
+
+		It("is rejected if it does not provide enough resources", func() {
+			tinyInstancetype := createInstancetype(2, "tiny-instancetype", "64M")
+			vm := randomVMWithInstancetype(tinyInstancetype.Name)
+
+			clusterPreferences, err := virtClient.VirtualMachineClusterPreference().List(context.Background(), metav1.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, preference := range clusterPreferences.Items {
+				vm.Name = fmt.Sprintf("test-vm-%s", preference.Name)
+				vm.Spec.Preference = &v1.PreferenceMatcher{
+					Name: preference.Name,
+				}
+
+				_, err := virtClient.VirtualMachine(testNamespace).Create(context.Background(), vm)
+				Expect(err).To(HaveOccurred())
+				Expect(err).To(MatchError(fmt.Sprintf(
+					"admission webhook \"virtualmachine-validator.kubevirt.io\" denied the request: "+
+						"failure checking preference requirements: "+
+						"insufficient Memory resources of 64M provided by instance type, preference requires %s",
+					preference.Spec.Requirements.Memory.Guest.String(),
+				)))
+			}
+			Expect(virtClient.VirtualMachineClusterInstancetype().Delete(context.Background(), "tiny-instancetype", metav1.DeleteOptions{})).ToNot(HaveOccurred())
```

Comment:
How do we want to handle cleanup of resources in general? How do other test suites do it?